lib/Doctrine/DBAL/Connection.php

```diff
@@ -435,6 +435,13 @@ private function getDatabasePlatformVersion()
             return $this->_conn->getServerVersion();
         }
 
+        if ($this->_conn instanceof \PDO && $this->_conn->getAttribute(PDO::ATTR_DRIVER_NAME)==='mysql') {
+            try {
+                return $this->_conn->query('select version()')->fetchColumn();
+            } catch (\PDOException $e) {
```

Comment:
It makes sense to make tests for my own code.  My problem was that when wrapping a PDO Connection of MySql , `getDatabasePlatformVersion` returns null. Looking more into the code I see there is a `requiresQueryForServerVersion` mechanism but for PDO this method return false. 

I'll dig more into this problem.